File: app/src/main/java/com/lazysoul/kotlinwithandroid/injection/module/ActivityModule.java
package com.lazysoul.kotlinwithandroid.injection.module;

import com.lazysoul.kotlinwithandroid.injection.scopes.ActivityScope;

import android.app.Activity;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Lazysoul on 2017. 7. 17..
 */

@Module
public class ActivityModule {

    private Activity activity;

    public ActivityModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    Activity provideActivity() {
        return activity;
    }
}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/injection/module/ActivityModule.kt
package com.lazysoul.kotlinwithandroid.injection.module

import android.app.Activity
import com.lazysoul.kotlinwithandroid.injection.scopes.ActivityScope
import dagger.Module
import dagger.Provides

/**
 * Created by Lazysoul on 2017. 7. 17..
 */

@Module
class ActivityModule(private val activity: Activity) {

    @Provides
    @ActivityScope
    internal fun provideActivity(): Activity = activity
}
